pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go

```diff
@@ -881,27 +881,15 @@ func (gce *GCECloud) firewallObject(name, region, desc string, sourceRanges nets
 }
 
 func (gce *GCECloud) projectOwnsStaticIP(name, region string, ipAddress string) (bool, error) {
-	pageToken := ""
-	page := 0
-	for ; page == 0 || (pageToken != "" && page < maxPages); page++ {
-		listCall := gce.service.Addresses.List(gce.projectID, region)
-		if pageToken != "" {
-			listCall = listCall.PageToken(pageToken)
-		}
-		addresses, err := listCall.Do()
-		if err != nil {
-			return false, fmt.Errorf("failed to list gce IP addresses: %v", err)
-		}
-		pageToken = addresses.NextPageToken
-		for _, addr := range addresses.Items {
-			if addr.Address == ipAddress {
-				// This project does own the address, so return success.
-				return true, nil
-			}
-		}
+	addrs, err := gce.service.Addresses.List(gce.projectID, region).Filter(fmt.Sprintf("address eq %s", ipAddress)).Do()
```

Comment:
On a side note, ideally we don't call the compute library directly here but instead add a func to gce_addresses.go.  This would be to capture metrics and support further unit testing. Thoughts?